Added support for all data types

alr2413 opened the pull request:
Replaced the string/model constructors with the generic type.

asetayes commented:
Hi @ericferreira1992,
I made some changes let me know if there is any problem with my changes

ericferreira1992 commented:
Hello @asetayes and @alr2413! Thanks for your contributions. I will take a time to analyze better these changes, and say if it really make sense for the package. Can you explain better (in detail) why the changes? (It would be a great)

alr2413 commented:
Thanks for your reply.
Instead of having a default constructor for both string and model, I changed it to have a generic constructor (T type) and by using that you'll be able to define your type: `AlphabetSearchView<T>.list`, The sorting functionality still are based on the title field of AlphabetSearchModel.
Also, I added one generic attribute to keep the data model in the Alpha class, so you can access all of data of that class in your builder function, i.e. `item.data`

alr2413 commented:
@ericferreira1992,
sorry I used two different accounts :)
let me know if you have any more question regarding these changes.

ericferreira1992 commented:
@alr2413,
I agree with your changes and improvements. But that changes will causes a breaking change, and if somebody is using it, it will stop working for it. Maybe we need to change version to 2.0.0, do you have some suggestion on it?

alr2413 commented:
Yes, you're right, I'll do that.

alr2413 committed:
Added generic type

alr2413 committed:
Update .gitignore

alr2413 committed:
fixed sorting issue

alr2413 committed:
Updated Readme

Upgraded all dependencies
fixed deprecated warning issue

alr2413 commented:
@ericferreira1992,
I updated library version, dependencies list, readme, changelog file.
let me know if there is any issue

ericferreira1992 commented:
@alr2413,
Done! I'll release that new version soon. Thanks a lot!

alr2413 commented:
@ericferreira1992,
Thanks for merging my changes to master.
just a reminder: you haven't published a release in `pub.dev` yet.

ericferreira1992 commented:
@alr2413,
I've just publish it. The package is in pending analysis now. 🙂
